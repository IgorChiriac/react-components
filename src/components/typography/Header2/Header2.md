```js
<Header2>Header2</Header2>
```

```js
<Header2 style={{ textAlign: 'center', color: 'salmon' }}>Header2</Header2>
```
