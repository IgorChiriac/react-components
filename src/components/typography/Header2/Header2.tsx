import React from 'react';
import styled from 'styled-components';
import * as fonts from '../../../constants/fonts';

export interface IHeader2Props extends React.HTMLAttributes<HTMLHeadingElement> {}

const SHeader = styled.h2`
  font-weight: normal;
  font-size: 32px;
  line-height: 48px;
  letter-spacing: 0;
  font-family: ${fonts.alverata};
`;

const Header2 = (props: IHeader2Props) => {
  return <SHeader {...props} />;
};

export default Header2;
