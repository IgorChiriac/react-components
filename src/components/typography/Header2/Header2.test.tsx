import React from 'react';
import { render } from 'react-testing-library';
import Header2 from './Header2';

describe('<Header2 />', () => {
  it('renders the component without props', () => {
    const { container } = render(<Header2>I'm an h2 header</Header2>);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders the component with props', () => {
    const { container } = render(<Header2 style={{ textAlign: 'center' }}>I'm an h2 header</Header2>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
