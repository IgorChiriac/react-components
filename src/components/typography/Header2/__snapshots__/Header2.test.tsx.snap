// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header2 /> renders the component with props 1`] = `
.c0 {
  font-weight: normal;
  font-size: 32px;
  line-height: 48px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Alverata,"Arial Black",Verdana,sans-serif;
}

<h2
  class="c0"
  style="text-align: center;"
>
  I'm an h2 header
</h2>
`;

exports[`<Header2 /> renders the component without props 1`] = `
.c0 {
  font-weight: normal;
  font-size: 32px;
  line-height: 48px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Alverata,"Arial Black",Verdana,sans-serif;
}

<h2
  class="c0"
>
  I'm an h2 header
</h2>
`;
